### PR TITLE
Replace string literals of localhost with function returning config

### PR DESCRIPTION
**Pull request by datamel**
These changes close https://github.com/cylc/cylc-flow/issues/4060

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).

- [x] Appropriate tests are included (unit and/or functional).
- [x] Already covered by existing tests.
- [x] No change log entry required (invisible to users).
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

File: cylc/flow/platforms.py [hunk 1]
```diff
@@ -481,3 +481,9 @@ def get_random_platform_for_install_target(install_target):
     """Return a randomly selected platform (dict) for given install target."""
     platforms = get_all_platforms_for_install_target(install_target)
     return random.choice(platforms)
+
+
+def get_localhost_install_target():
+    """Returns the install target of localhost platform"""
+    localhost = get_platform('localhost')
```
Comment (wxtim) [inline cylc/flow/platforms.py:488]: Is identical to:
```suggestion
    localhost = get_platform()
```

File: cylc/flow/platforms.py [hunk 2]
```diff
@@ -481,3 +481,9 @@ def get_random_platform_for_install_target(install_target):
...
+    return get_install_target_from_platform(localhost)
```
Comment (wxtim) [inline cylc/flow/platforms.py:489]: I've tried to kill VSCode's line stripping behavior, but it seems to have become more aggressive. Or perhaps I'm creating more new files....
```suggestion
    return get_install_target_from_platform(localhost)

```

File: tests/unit/test_platforms_get_platform.py [hunk 3]
```diff
@@ -203,3 +206,7 @@ def test_get_platform_warn_mode_fail_if_backticks():
         r'platform = `echo \$\{chamber\}`: '
         r'backticks are not supported; please use \$\(\)'
     )
+
+
+def test_get_localhost_install_target():
+    assert get_localhost_install_target() == 'localhost'
```
Comment (wxtim) [inline tests/unit/test_platforms_get_platform.py:212]:
```suggestion
    assert get_localhost_install_target() == 'localhost'

```

File: tests/unit/test_platforms_get_platform.py [hunk 4]
```diff
@@ -203,3 +206,7 @@ def test_get_platform_warn_mode_fail_if_backticks():
         r'platform = `echo \$\{chamber\}`: '
         r'backticks are not supported; please use \$\(\)'
     )
+
+
+def test_get_localhost_install_target():
```
Comment (wxtim) [inline tests/unit/test_platforms_get_platform.py:211]: Might you be able to model the case where
```
[platforms]
  [[localhost]]
    install target = raspi123
```

**Review by wxtim — CHANGES_REQUESTED** [review]
- [x] Read code, which seems to make sense. I have jotted some thoughts about the code, but there's nothing critical in the comments, except possibly `get_platform('localhost') which might just possibly later cause future inconsistencies.
- [x] Read Tests - Suggested 1 additional test scenario.
- [x] Downloaded and tested locally - will do.

File: tests/unit/test_platforms_get_platform.py (hunk 3 above)
Comment (datamel) [reply tests/unit/test_platforms_get_platform.py:212]: Ah, I think this is to do with the way the diff is displayed on github. Blank line 213 was there, just not displayed as it was no change to the original?

File: cylc/flow/platforms.py (hunk 1 above)
Comment (datamel) [reply cylc/flow/platforms.py]: I changed it, making the default of  `get_install_target_from_platform()` return localhost but then I changed it back as I think  `if platform['install target'] == get_localhost_install_target()` immediately makes more sense for future devs than `if platform['install target'] == get_install_target_from_platfrom() `, what do you think?

Comment (wxtim) [reply cylc/flow/platforms.py]: It's fine. My suggestion only makes sense in context of platforms. Yours just makes sense.

**Review by wxtim — APPROVED** [review]

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**